### ISSUE TITLE
[Bug?]: unclosed-tags inside of innerHTML result in hydration mismatch

### ISSUE THREAD

**Issue by indeyets** [solidjs/solid-start#1411]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

code like this is guaranteed to cause hydration mismatch `<p innerHTML={'<a href="http://example.com">test'} />`

### Expected behavior 🤔

The code is injected and is equivalent to `<p><a href="http://example.com">test</p>` and there is no error.

### Steps to reproduce 🕹

Steps:

1. checkout https://github.com/indeyets/solid-start-bug-innerhtml and install dependencies
2. run `npm run dev`
3. open http://localhost:3000/ in browser

**Comment by ryansolid** [comment]
This is an odd one in that we skip innerHTML on hydration altogether. Or rather we should be. And it doesn't use JSX so no hydration IDs. It shouldn't have any bearing on hydration even if we didn't,.... unless the insert on the server is the problem and it produces malformed HTML.

I do remember from my research on omitting tags some tags had really weird rules and `<a>` were on a list I had to always close otherwise they produce really weird behavior, like looking for the next `<a>` or something like that. I think I actually see it in your demo, the hello world becomes underlined for a moment suggesting it is a link before the hydration error pops up.

Yeah basically any loose text after it becomes wrapped in `<a href="example.com"></a>`. Wo while over the wire the only difference between including and not including the closing tag is the missing `</a>`, the browser heals it in a way that inserts a bunch of extra anchor tags.

Long story short close your anchor tags, as native browser behavior is odd. I recommend closing tags in general unless you are comfortable with being aware of all the weird behaviors that happen when you don't.

**Comment by indeyets** [comment]
@ryansolid is there any chance to get a controllable error in this case? that would be fine with me.

my use-case involves input from a third-party system. and while I can validate what I get to the certain degree, I can't be 100% sure that I did it properly.

So, I need a way to handle this issue at this exact location. I'm fine with showing an error message instead of the block or even with hiding this block altogether. But I need my app to keep working otherwise
